pkg/cgu/cgu.go

```diff
@@ -338,79 +336,55 @@ func (builder *CguBuilder) WaitUntilDeleted(timeout time.Duration) error {
 		})
 }
 
-// validate will check that the builder and builder definition are properly initialized before
-// accessing any member fields.
-func (builder *CguBuilder) validate() (bool, error) {
-	resourceCRD := "cgu"
-
-	if builder == nil {
-		glog.V(100).Infof("The %s builder is uninitialized", resourceCRD)
-
-		return false, fmt.Errorf("error: received nil %s builder", resourceCRD)
-	}
-
-	if builder.Definition == nil {
-		glog.V(100).Infof("The %s is undefined", resourceCRD)
-
-		return false, fmt.Errorf(msg.UndefinedCrdObjectErrString(resourceCRD))
-	}
-
-	if builder.apiClient == nil {
-		glog.V(100).Infof("The %s builder apiclient is nil", resourceCRD)
-
-		return false, fmt.Errorf("%s builder cannot have nil apiClient", resourceCRD)
-	}
-
-	if builder.errorMsg != "" {
-		glog.V(100).Infof("The %s builder has error message: %s", resourceCRD, builder.errorMsg)
-
-		return false, fmt.Errorf(builder.errorMsg)
-	}
-
-	return true, nil
-}
-
-// WaitUntilComplete waits the specified timeout for the CGU to complete.
-func (builder *CguBuilder) WaitUntilComplete(timeout time.Duration) (*CguBuilder, error) {
+// WaitForCondition waits until the CGU has a condition that matches the expected, checking only the Type, Status,
+// Reason, and Message fields. For the message field, it matches if the message contains the expected. Zero fields in
+// the expected condition are ignored.
+func (builder *CguBuilder) WaitForCondition(expected metav1.Condition, timeout time.Duration) (*CguBuilder, error) {
 	if valid, err := builder.validate(); !valid {
-		return builder, err
+		return nil, err
```

Comment:
From what I see, only in `Get()` functions we return the nil builder. Wonder why you want to do that in a `Wait()` function

Reply:
@cdvultur I think changing to return nil and an error is one of the goals of #355

Reply:
Please revert it back to:
`return builder, err
`